mesonbuild/dependencies/mpi.py

```diff
@@ -40,41 +41,43 @@ def mpi_factory(env: 'Environment',
     if DependencyMethods.CONFIG_TOOL in methods:
         nwargs = kwargs.copy()
 
+        # We try the environment variables for the tools first, but then
+        # fall back to the hardcoded names
+
+        if language == 'c':
+            env_vars = ['MPICC']
+        elif language == 'cpp':
+            env_vars = ['MPICXX']
+        elif language == 'fortran':
+            env_vars = ['MPIFC', 'MPIF90', 'MPIF77']
+
+        tool_names = [os.environ.get(env_name) for env_name in env_vars]
+        tool_names = [t for t in tool_names if t]  # remove empty environment variables
+
         if compiler_is_intel:
             if env.machines[for_machine].is_windows():
-                nwargs['version_arg'] = '-v'
                 nwargs['returncode_value'] = 3
 
             if language == 'c':
-                tool_names = [os.environ.get('I_MPI_CC'), 'mpiicc']
+                tool_names.append('mpiicc')
             elif language == 'cpp':
-                tool_names = [os.environ.get('I_MPI_CXX'), 'mpiicpc']
+                tool_names.append('mpiicpc')
             elif language == 'fortran':
-                tool_names = [os.environ.get('I_MPI_F90'), 'mpiifort']
+                tool_names.append('mpiifort')
```

Comment:
So we no longer look up these `I_MPI_*` variables at all, just `MPI*` variables. I wonder where the original variables came from...

Reply:
These are environment variables used internally by mpiicc (https://www.intel.com/content/www/us/en/docs/mpi-library/developer-reference-windows/2021-8/compilation-environment-variables.html).

> Set this environment variable to select a specific compiler to be used. Specify the full path to the compiler if it is not located in the search path.